library/Garp/Auth/Adapter/Passwordless.php

```diff
@@ -217,22 +217,22 @@ protected function _tokenIsExpired(Garp_Db_Table_Row $row) {
     }
 
     protected function _getExpirationDate() {
-        if ($this->_getAuthVars() && array_key_exists('token_expires_in', $this->_getAuthVars())) {
-            $authVars = $this->_getAuthVars();
-            return date('Y-m-d H:i:s', strtotime($authVars['token_expires_in']));
-        }
-        return date('Y-m-d H:i:s', strtotime(self::DEFAULT_TOKEN_EXPIRATION_TIME));
+        return date(
+            'Y-m-d H:i:s',
+            strtotime($this->_getAuthVars()->token_expires_in ?? self::DEFAULT_TOKEN_EXPIRATION_TIME)
+        );
     }
 
     protected function _sendTokenEmail($email, $userId, $token) {
-        $mailer = new Garp_Mailer();
-        return $mailer->send(
-            array(
+        $params = class_exists('App_LoginTokenMailer')
+            ? (new App_LoginTokenMailer($email, $userId, $this->_getLoginUrl($userId, $token)))->getEmailParams()
+            : [
                 'to' => $email,
                 'subject' => $this->_getEmailSubject(),
-                'message' => $this->_getEmailBody($userId, $token)
-            )
-        );
+                'message' => $this->_getEmailBody($userId, $token),
+            ];
+        $mailer = new Garp_Mailer();
+        return $mailer->send($params);
```

Comment:
Ik zou deze functie nog iets rustiger vinden als je indien `App_LoginTokenMailer` niet bestaat je een `Garp_LoginTokenMailer` gebruikt. Het maakt niet zoveel uit dat het maar een klein functietje is. 
Dan kun je ze ook een interface laten delen en weet je waar `App_LoginTokenMailer` aan moet voldoen.

Als deze class misschien een `_getLoginTokenMailer` functie heeft met als signature `: BlaBla_LoginTokenMail_Interface` dan is dat meteen gecontroleerd.

Reply:
Ik zou deze functie één object meegeven. Met een interface die Harmen suggereert. Dan kan je bij de aanroep van deze functie zelf een object in elkaar zetten.

Volgens mij is het ook duidelijker om`getEmailParams` te vervangen door drie getters. Dan is gelijk duidelijk wat de structuur van de data is. Nu kan alles in de array gepropt worden en kunnen verschillende implementaties niet compatible zijn.

Reply:
Ik heb de aanpassingen gedaan. Ik moet alleen nog een oplossing hebben voor het feit dat ik niet weet ik of `htmlMessage` data moet versturen of `message`. Zien jullie hier een oplossing voor?